src/ports/transaction/component.ts

```diff
@@ -1,120 +1,60 @@
-  MetaTransactionRequest,
-  MetaTransactionResponse,
-  MetaTransactionStatus,
-  TransactionData,
-  TransactionRow,
-} from './types'
+import { ITransactionComponent, TransactionRow } from './types'
 
 export function createTransactionComponent(
   components: Pick<
     AppComponents,
-    'config' | 'pg' | 'contracts' | 'features' | 'fetcher' | 'logs' | 'metrics'
+    | 'config'
+    | 'pg'
+    | 'contracts'
+    | 'features'
+    | 'fetcher'
+    | 'logs'
+    | 'metrics'
+    | 'gelato'
+    | 'biconomy'
   >
 ): ITransactionComponent {
-  const { config, pg, fetcher, metrics } = components
+  const { features, gelato, biconomy, pg } = components
 
   async function sendMetaTransaction(
     transactionData: TransactionData
   ): Promise<string> {
-    const biconomyAPIId = await config.requireString('BICONOMY_API_ID')
-    const biconomyAPIKey = await config.requireString('BICONOMY_API_KEY')
-    const biconomyAPIURL = await config.requireString('BICONOMY_API_URL')
-
-    const body: MetaTransactionRequest = {
-      apiId: biconomyAPIId,
-      ...transactionData,
-    }
-
-    const result = await fetcher.fetch(biconomyAPIURL, {
-      headers: {
-        'x-api-key': biconomyAPIKey,
-        'content-type': 'application/json',
-      },
-      body: JSON.stringify(body),
-      method: 'POST',
-    })
-
-    if (result.status !== MetaTransactionStatus.OK) {
-      let message: string | undefined
-      let code: ErrorCode | undefined
-
-      switch (result.status) {
-        case MetaTransactionStatus.CONFLICT:
-          const response: MetaTransactionResponse = await result.json()
-          // Conflict errors always have message and code values
-          message = response.message!
-          code = toErrorCode(response.code!)
-
-          // A limit was reached, check ErrorCode for possible values
-          metrics.increment('dcl_error_limit_reached_transactions_biconomy', {
-            code,
-          })
-          break
-        case MetaTransactionStatus.EXPECTATION_FAILED:
-          code = ErrorCode.EXPECTATION_FAILED
-
-          // This error happens when the contract execution will fail. See https://github.com/decentraland/transactions-server/blob/2e5d833f672a87a7acf0ff761f986421676c4ec9/ERRORS.md
-          metrics.increment(
-            'dcl_error_cannot_estimate_gas_transactions_biconomy'
-          )
-          break
-        case MetaTransactionStatus.NOT_FOUND:
-        case MetaTransactionStatus.INTERNAL_SERVER_ERROR:
-        default:
-          // Any other error is related to the Biconomy API
-          metrics.increment('dcl_error_relay_transactions_biconomy')
-          break
-      }
-
-      message = message || (await result.text())
-
-      throw new InvalidTransactionError(
-        `An error occurred trying to send the meta transaction. Response: ${message}.
-          ${result.statusText}`,
-        code
-      )
+    const isGelatoRelayerEnabled = await features.getIsFeatureEnabled(
```

Comment:
a bit repeatitive but this could also be defined outside the fn right? to change the FF we need to redeploy anyways, this won't change between `sendMetaTransaction` calls

Reply:
This can't be moved outside because it's actually using the feature flag to fetch the feature flag. The features module fetches the flag each time it is requested if there's no environment variable that defines it.